drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1745,6 +1747,11 @@ static int uarte_instance_init(const struct device *dev,
 
 	data->dev = dev;
 
+#ifdef CONFIG_ARCH_POSIX
+	/* For simulation the DT provided peripheral address needs to be corrected */
+	((struct pinctrl_dev_config *)cfg->pcfg)->reg = (uintptr_t)cfg->uarte_regs;
```

Comment:
Is ARCH_POSIX always considered simulation?

Reply:
"Yes" :). I mean this driver will be compiled for CONFIG_ARCH_POSIX only when built for the nrf*_bsim boards.
And in any case we are calling the new native_sim "simulator" too.. I guess it is a matter of convention.
(Wrt the need to replace registers addresses from hardcoded values to wherever they happen to be placed, that exists today for all posix arch targets)